File: LeetCode/python3/933.py
class RecentCounter:

    def __init__(self):
        self.time = []
        self.i, self.j = 0, 0

    def ping(self, t):
        """
        :type t: int
        :rtype: int
        """
        self.j += 1
        self.time.append(t)
        while self.i < self.j:
            if self.time[self.j - 1 ] - self.time[self.i] <= 3000:
                break
            self.i += 1
        return self.j-self.i

File: LeetCode/cpp/933.cpp
class RecentCounter {
public:
    vector<int> time;
    int i, j;

    RecentCounter() {
        time = vector<int>();
        i = 0;
        j = 0;
    }

    int ping(int t) {
        ++j;
        time.push_back(t);
        while (i < j) {
            if (time[j - 1] - time[i] <= 3000)
                break;
            ++i;
        }
        return j - i;
    }
};
